File: 867-transpose-matrix/867-transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        m, n = len(matrix), len(matrix[0])
        new = [[0 for i in range(m)] for j in range(n)]
        for i in range(n):
            for j in range(m):
                new[i][j] = matrix[j][i]
        return new

File: 867-transpose-matrix/867-transpose-matrix.go
func transpose(matrix [][]int) [][]int {
    var m, n int
    m = len(matrix)
    n = len(matrix[0])
    transpose := make([][]int, n)
    //var transpose [n][m]int
    /*for i := 0; i<m ; i++ {
        transpose[i] = make([]int, m)
    }*/
    for i := 0; i<m ; i++ {
        //transpose[i] = make([]int, m)
        for j := 0; j<n ; j++ {
            //transpose[j][i] = matrix[i][j]
            transpose[j] = append(transpose[j], matrix[i][j])
        }
    }
    return transpose
}